frontend/src/metabase/visualizations/components/ChartSettings.jsx

```diff
@@ -355,8 +391,14 @@ class ChartSettings extends Component {
           </div>
         )}
         <ChartSettingsWidgetPopover
+          currentWidgetKey={
+            currentWidget?.props?.initialKey || currentWidget?.props?.seriesKey
+          }
           anchor={popoverRef}
-          widget={getPopoverWidget(widgets, currentWidget, extraWidgetProps)}
+          widgets={[
+            this.getFormattingWidget(extraWidgetProps),
+            this.getStyleWidget(),
+          ].filter(x => !!x)}
```

Comment:
We can often see `.filter(Boolean)` in our codebase as well. However, the downside is that Typescript does not understand that so if an array had a type `Something | undefined` it will remain so after filtering. It would be better to use something like https://github.com/metabase/metabase/pull/25408/files#diff-9fe025708714336edb9ea4f75faee7fd261a3e62fbd0b558460d4449990be730R1 but it has not been merged yet

Reply:
Wow I'm thinking about something like that. Good to know that there's a `isNotEmpty` function that I could use.